### PR TITLE
Remove unused Post Editor code

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR removes a lot of unused code from the Post Editor. I'm doing this cleanup to make the Redux migration easier.

Best reviewed when going through the individual commits and reading the commit messages to learn more details.

The main change is removing the `post-editor/revisions` config option that has been enabled in all environments for a few months.

Then there is `EditorVisibility` refactoring that moves retrieving of the `isPrivateSite` prop into the `connect` call, rather that passing it down from parent components.

The rest is removing obviously unused methods and props.

### CODE REVIEW

File: client/post-editor/editor-visibility/index.jsx [hunk 1]
```diff
@@ -78,16 +77,6 @@ class EditorVisibility extends React.Component {
 		return 'public';
 	};
 
-	isPasswordValid = () => {
```
Comment (blowery) [inline client/post-editor/editor-visibility/index.jsx]: this doesn't match the commit message?

**Comment by blowery** [comment]
All the commits look sane. Weird that all the tests failed...

**Review by blowery — APPROVED** [review]
LGTM, editor works as expected.

**Review by spen — APPROVED** [review]
Awesome work removing this redundant code @jsnajdr - thanks a bunch for this!

Editor & revisions working as expected :)

**Pull request merged**